### ISSUE TITLE
Allow skipped checks?

### ISSUE THREAD

**Issue by Billy-** [lewagon/wait-on-check-action#12]
Hi :wave: 

I have just added some jobs which get skipped on my master branch - and this has caused the wait-on-check action to fail :slightly_smiling_face: 

My initial thought is that the action could consider skipped checks a success. What do you think? If this doesn't seem like something you'd want to do, maybe we could add a parameter to control how you want it to behave? Something like `allowSkipped` or `allowStatuses: "success,skipped"` or something?

**Comment by progapandist** [comment]
I'll take a look into this, thanks for reporting!

**Comment by progapandist** [comment]
If you want to give it a try (I think having an `allow_skipped` flag and enabling it by default is the best UX here) and make a PR — we now have a nice test coverage by @matiasalbarello, and the "test" actions in this repo. Just LMK, if not — I'll take care of it myself when my hands a less full.

**Comment by AraHaan** [comment]
@progapandist what about the case where I got to like do the following:

- pull request build succeeds.
- pull request executes another workflow (automerge)
- pull request also executes another job in the build workflow (which should wait till the automerge workflow in the pull request succeeds and merge the pull request before doing anything in that job (this job checks for updates with ``git submodule update --remote`` then uses the create-pull-request action to create another pull request).

Currently without this it is possible to then notice it open double pull requests for the same exact changes it generates.

**Comment by matiasalbarello** [comment]
@progapandist can we make a release with the last changes that solve this issue so we can close it?

**Comment by matiasalbarello** [comment]
Sorry, closed by mistake.

**Comment by progapandist** [comment]
Sure! Can you please draft a quick CHANGELOG to describe everything we added since the last release and then I will bump it? Then we can keep adding to that CHANGELOG when/if we add more features. Let's call this one 0.3

**Comment by emmanuelvisage** [comment]
Hi, thanks for this github action, it's very useful.
What exactly qualifies as a 'skipped check' ? Because I have a check in a migration workflow that sometimes isn't triggered  in the run (because there is no new migration) and the wait on this check breaks with "The requested check was never run against this ref, exiting..." although I am on the master branch and it says it's checking on the "success,skipped" checks. 

Is this possible to continue when the check was not even found in the run ?

**Comment by progapandist** [comment]
@Billy- — you are the feature author, anything to add?

**Comment by matiasalbarello** [comment]
> Hi, thanks for this github action, it's very useful.
> What exactly qualifies as a 'skipped check' ? Because I have a check in a migration workflow that sometimes isn't triggered in the run (because there is no new migration) and the wait on this check in other workflows breaks with "The requested check was never run against this ref, exiting..." although I am on the master branch and it says it's checking on the "success,skipped" checks.
> 
> Is this possible to continue when the check was not even found in the run?

Hi @emmanuelvisage. Here you can find an example of a skipped job: https://github.com/lewagon/wait-on-check-action/blob/master/.github/workflows/wait-for-allowed-conclusions.yml
So, a skipped job actually has run (at least it belongs to a workflow triggered by some event) but finished with a status of `skipped`. 
If the workflow is not triggered (as you mentioned, for example, because a certain event didn't occur), and this action was configured to wait for it, it will fail as expected. There's no way to know if the action that didn't run was supposed to run or not.
If you omit the check-name, it will wait for all actions to run. If the workflow doesn't run, it would not wait for that one. 

Hope it helps and please don't hesitate to write me back if something is not yet clear.

**Comment by melyux** [comment]
If it doesn't find a job with the name specified, it should just move on, right? Maybe check a few times if it is actually indistinguishable from some other case. I don't want to clutter my Actions with skipped jobs that should ideally never trigger, except to work around this issue.
